src/main/java/com/petros/bibernate/datasource/BibernateConnectionPool.java

```diff
@@ -1,23 +1,29 @@
 package com.petros.bibernate.datasource;
 
 import com.petros.bibernate.exception.BibernateException;
+import lombok.extern.slf4j.Slf4j;
 
 import java.sql.Connection;
 import java.sql.DriverManager;
 import java.sql.SQLException;
-import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.Queue;
+import java.util.concurrent.LinkedBlockingQueue;
 
+@Slf4j
 public class BibernateConnectionPool {
-    private final ConcurrentLinkedQueue<BibernateConnection> pool;
-    private final int CONNECTION_MAX_SIZE = 10;
+    private final Queue<BibernateConnection> pool;
 
-    public BibernateConnectionPool(String url, String username, String password) {
-        pool = new ConcurrentLinkedQueue<>();
-        for (int i = 0; i < CONNECTION_MAX_SIZE; i++) {
+    public BibernateConnectionPool(String url, String username, String password, int connectionPoolSize) {
+        log.info("Connection pool is going to be created");
+        pool = new LinkedBlockingQueue<>();
```

Comment:
Is LinkedBlockingQueue in Java thread-safe?

Reply:
Yes, it is thread-safe